daemon/start.go

```diff
@@ -224,6 +224,15 @@ func (daemon *Daemon) containerStart(ctx context.Context, daemonCfg *configStore
 		}
 	}()
 
+	// Finish sandbox initialisation (if there's a sandbox, so not using container networking).
+	if sb, err := daemon.netController.GetSandbox(container.ID); err == nil {
+		if err := sb.CtrTaskCreated(); err != nil {
+			log.G(ctx).WithError(err).WithField("container", container.ID).
+				Error("failed to complete network configuration")
+			return errors.Wrap(err, "failed to complete network configuration")
```

Comment:
We should probably return an appropriate `errdefs` error here (as we do for other errors returned). I guess this should be considered an internal server error (i.e., not a "user error"), in which case;


```suggestion
			return errdefs.System(errors.Wrap(err, "failed to complete network configuration"))
```

Reply:
The 'err' has been `errdefs.System()`'d by the time it gets here, in `Sandbox.ctrTaskCreatedDNS()`. The Wrap() was just to add a bit of context about what was going on at the time.

I did it that way thinking it might be best to leave the possibility of `CtrTaskCreated()` finding/returning a user/config error. (Maybe address allocation should happen at this point, after the IPv6-ness of the container is known. That might mean we can avoid allocating an IPv6 addresses from the pool if it won't be usable in case, for example, there's no address left in the pool. But, haven't thought that through properly yet, spitting the error out earlier for the common case where the container does have IPv6 might be best.)

So, could change it if you think that'd be best?

Reply:
oh! if it's already typed, then I guess we don't need to; sorry missed that 😅